@xen-orchestra/fs/src/s3.js

```diff
@@ -27,30 +41,28 @@ export default class S3Handler extends RemoteHandlerAbstract {
     super(remote)
     const { allowUnauthorized, host, path, username, password, protocol, region } = parse(remote.url)
     const params = {
-      accessKeyId: username,
       apiVersion: '2006-03-01',
-      endpoint: host,
-      s3ForcePathStyle: true,
-      secretAccessKey: password,
-      signatureVersion: 'v4',
-      httpOptions: {
-        timeout: 600000,
+      endpoint: `${protocol}://${host}`,
+      forcePathStyle: true,
+      credentials: {
+        accessKeyId: username,
+        secretAccessKey: password,
       },
-    }
-    if (protocol === 'http') {
-      params.httpOptions.agent = new http.Agent({ keepAlive: true })
-      params.sslEnabled = false
-    } else if (protocol === 'https') {
-      params.httpOptions.agent = new https.Agent({
-        rejectUnauthorized: !allowUnauthorized,
-        keepAlive: true,
+      tls: protocol === 'https',
+      region: region || 'us-east-1',
```

Comment:
region can be empty with some provider ( like minio )

Reply:
SDK v3 no longer accepts empty region.

The internal v2 behavior was to use 'us-east-1' as default region, according to [this issue](https://github.com/aws/aws-sdk-js-v3/issues/1845#issuecomment-754832210)